packages/router-plugin/tests/router.plugin.spec.ts

```diff
@@ -133,45 +155,348 @@ describe('NgxsRouterPlugin', () => {
+
+      // The very first `ResolveEnd` event is triggered during root module bootstrapping
+      actions$
+        .pipe(
+          ofActionSuccessful(RouterDataResolved),
+          first()
+        )
+        .subscribe(({ routerState }: RouterDataResolved) => {
+          // Assert
+          const dataFromTheEvent = routerState.root.firstChild!.data;
+          expect(dataFromTheEvent).toEqual({ test });
+        });
+
+      await store
+        .dispatch(
+          new Navigate(
+            ['/'],
+            {
+              a: 10
+            },
+            {
+              queryParamsHandling: 'merge'
+            }
+          )
+        )
+        .toPromise();
+
+      resetPlatformAfterBootstrapping();
+    });
+
+    it('should update the state if navigation is performed between the same component', async () => {
+      // Arrange
+      destroyPlatformBeforeBootstrappingTheNewOne();
+
+      @State({
+        name: 'counter',
+        defaults: 0
+      })
+      class CounterState {
+        @Action(RouterNavigation)
+        public routerNavigation({ getState, setState }: StateContext<number>): void {
+          setState(getState() + 1);
+        }
+      }
+
+      // Act
+      const { injector } = await platformBrowserDynamic().bootstrapModule(
+        getTestModule([CounterState])
+      );
+      const store: Store = injector.get(Store);
+      const router: Router = injector.get(Router);
+
+      await router.navigateByUrl('/a/b/c');
+      await router.navigateByUrl('/a/b');
+
+      // Assert
+      const counter = store.selectSnapshot<number>(CounterState);
+      expect(counter).toEqual(3);
+
+      resetPlatformAfterBootstrapping();
+    });
+
+    it('should call selector if navigation is performed between the same component', async () => {
+      // Arrange
+      destroyPlatformBeforeBootstrappingTheNewOne();
+
+      let selectorCalledTimes = 0;
+
+      @State({
+        name: 'counter',
+        defaults: 0
+      })
+      class CounterState {
+        @Selector()
+        public static counter(state: number) {
+          selectorCalledTimes++;
+          return state;
+        }
+
+        @Action(RouterNavigation)
+        public routerNavigation({ getState, setState }: StateContext<number>): void {
+          setState(getState() + 1);
+        }
+      }
+
+      // Act
+      const { injector } = await platformBrowserDynamic().bootstrapModule(
+        getTestModule([CounterState])
+      );
+      const store: Store = injector.get(Store);
+      const router: Router = injector.get(Router);
+
+      store
+        .select(CounterState.counter)
+        .pipe(
+          filter(() => selectorCalledTimes === 3),
+          take(1)
+        )
+        .subscribe(counter => {
+          // Assert
+          expect(selectorCalledTimes).toEqual(3);
+          expect(selectorCalledTimes).toEqual(counter);
+        });
+
+      await router.navigateByUrl('/a/b/c');
+      await router.navigateByUrl('/a/b');
```

Comment:
Probably better to have:
```
      const subscription = store
        .select(CounterState.counter)
        .subscribe(counter => { });

      await router.navigateByUrl('/a/b/c');
      await router.navigateByUrl('/a/b');
      subscription.unsubscribe();

      // Assert
      expect(selectorCalledTimes).toEqual(3);
      expect(selectorCalledTimes).toEqual(counter);
```
Thus the expects would be guaranteed to be called.
In your code it may silently fail if the selectors are called less than three times and would pass if called more than three times.